e2e_test/batch/types/array_ty.slt.part

```diff
@@ -105,3 +105,118 @@ drop table t;
 # This is different from PostgreSQL, we may want to change this in the future.
 statement ok
 select array[array[1,2], array[3]];
+
+# array range access
+query T
+select array[1,NULL,2][-1:134124523];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[3:1];
+----
+{}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523];
+----
+{{1},{2},{3}}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1];
+----
+{1}
+
+query T
+select array[array[1],array[2],array[3]][-21432315:134124523][1:2];
+----
+{{1},{2}}
+
+query I
+select array[array[1],array[2],array[3]][-21432315:134124523][1][1];
+----
+1
+
+query T
+select (array[1,NULL,2])[:3];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[:2];
+----
+{1,NULL}
+
+query T
+select (array[1,NULL,2])[:1];
+----
+{1}
+
+query T
+select (array[1,NULL,2])[:999];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[:0];
+----
+{}
+
+query T
+select (array[1,NULL,2])[:-1];
+----
+{}
+
+query T
+select (array[1,NULL,2])[-1:];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[0:];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[1:];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[2:];
+----
+{NULL,2}
+
+query T
+select (array[1,NULL,2])[3:];
+----
+{2}
+
+query T
+select (array[1,NULL,2])[4:];
+----
+{}
+
+query T
+select (array[1,NULL,2])[5:];
+----
+{}
+
+query T
+select (array[1,NULL,2])[:];
+----
+{1,NULL,2}
+
+query T
+select (array[1,NULL,2])[5:-1];
+----
+{}
+
+query T
+select (array[1,NULL,2])[2:1];
+----
+{}
+
+# index larger than int32
+statement error db error: ERROR: QueryError: Expr error: Unsupported function: Ok\(ArrayRangeAccess\)
```

Comment:
It seems the implicit cast to `int32` is not working as expected?
I expect the cast will enable this case